tt_metal/host_api.hpp

```diff
@@ -56,11 +56,11 @@ size_t GetNumPCIeDevices();
  *
  * Return value: Device *
  *
- * | Argument  | Description                | Type            | Valid Range                       | Required |
- * |-----------|----------------------------|-----------------|-----------------------------------|----------|
- * | device_id | ID of the device to target | chip_id_t (int) | 0 to (GetNumAvailableDevices - 1) | Yes      |
+ * | Argument   | Description                | Type            | Valid Range                       | Required |
+ * |------------|----------------------------|-----------------|-----------------------------------|----------|
+ * | device_id  | ID of the device to target| chip_id_t (int) | 0 to (GetNumAvailableDevices - 1) | Yes      |
  * */
-Device *CreateDevice(chip_id_t device_id, const std::vector<uint32_t>& l1_bank_remap = {});
+Device *CreateDevice(chip_id_t device_id, const uint8_t num_hw_cqs = 1, const std::vector<uint32_t>& l1_bank_remap = {});
```

Comment:
the yaml contains all the possible configurations. we need to specify which config we want since there is compile time information inferred based on the num of hw cqs

Reply:
also will user be able to put any number of HW CQs as a parameter (3, 4, 10, 100)? If not, this level of configurability shouldn't be exposed

Reply:
is it possible to use only the 2 HW CQ yaml config from here on out? You'd need to be backwards compatiable with user requesting only 1 queue. This would remove the need for configuring it here and also remove need for a separate GTest test suite

Reply:
After re-reading your high-level PR description, we should try to address this in the follow-on PR